src/tactic/core.lean

```diff
@@ -353,6 +344,62 @@ do to_remove ← hs.mfilter $ λ h, do {
   to_remove.mmap' (λ h, try (clear h)),
   return (¬ to_remove.empty ∨ goal_simplified)
 
+/-- `revert_after e` reverts all local constants after local constant `e`. -/
+meta def revert_after (e : expr) : tactic ℕ := do
+  l ← local_context,
+  [pos] ← return $ l.indexes_of e | pp e >>= λ s, fail format!"No such local constant {s}",
+  let l := l.drop pos.succ, -- all local hypotheses after `e`
+  revert_lst l
+
+/-- `generalize' e n` is similar as `generalize` but also succeeds when `e` does not occur in the
```

Comment:
```suggestion
/-- `generalize' e n` is similar to `generalize` but also succeeds when `e` does not occur in the
```
Unfortunately the doc string for `tactic.generalize` isn't helpful at all. Could you expand this a bit to explain what it does, without referring to the old `generalize`?